src/test/groovy/org/prebid/server/functional/testcontainers/scaffolding/NetworkScaffolding.groovy

```diff
@@ -46,6 +47,21 @@ abstract class NetworkScaffolding {
                         .size()
     }
 
+    boolean checkRequestCount(int expectedCount, int pollTime = 1000) {
+        def frequencyPolling = 10
+        def isRequestReceived = false
+
+        for (int i = 0; i < pollTime; i += frequencyPolling) {
```

Comment:
Please write proper poller. If operation inside this loop will take more than `pollTime`, you will still continue looping.